mlir/test/Target/LLVMIR/genx.mlir

```diff
@@ -135,8 +135,15 @@ llvm.func @genx.atomic.rmw(%ptr : !llvm.ptr<i32, 1>, %sptr : !llvm.ptr<i64, 3>,
   llvm.return
 }
 
+llvm.func @genx.dpas(%c : vector<8xi32>, %a : vector<8xi32>, %b : vector<8xi32>) {
+  // CHECK: %4 = call <8 x i32> @llvm.genx.GenISA.dpas.v8i32.v8i32.v8i32.v8i32(<8 x i32> %0, <8 x i32> %1, <8 x i32> %2, i32 32, i32 4, i32 1, i32 1, i1 false)
+  %0 = genx.matrix.dpas %c, %a, %b {pa=32:i32, pb=4:i32, sd=1:i32, rc=1:i32, isDpasw=false} : (vector<8xi32>, vector<8xi32>, vector<8xi32>) -> vector<8xi32>
```

Comment:
I think the elem. type of %a and %b has to be i8 when C has elem. type i32. Anyway once the verification code for the operation is added illegal combination of elem. types and attribute values should be diagnosed.

Reply:
It is probably better to use f32 type and PVC parameters for the test case to illustrate its usage: %c : vector<8x16xf32>, %a : vector<8x8xf32>, %b: vector<8x16xf32>, pa=32, pb=32, sd=8, rc=8

Reply:
Maybe even %b: vector<16x8xf32> for the column major layout.

Reply:
multi-dimensional vector is not supported in LLVM.

Reply:
Let's keep this test case for now, having difficulty to understand what we should allowed on PVC. We can change it when we understand better.

Reply:
Okey.  We can probably linearize the 2D submatrix to a 1-D vector: %c: vector<128xf32>, %a: vector<64xf32>, %b<128xf32>. Let's do it later then.

Reply:
From the verification code https://github.com/intel-innersource/drivers.gpu.unified/blob/comp/igc/Source/IGC/Compiler/Optimizer/OpenCLPasses/DpasFuncs/DpasFuncsResolution.cpp#L457, number of elements should be equal to $rc or $sd. However, $rc should be equals to M. so seems like we should not linearize the 2D submatrix.